### PR TITLE
KTOR-7846 Fix running tests against JDK 8

### PR DESCRIPTION
**Pull request by osipxd**
**Subsystem**
Test Infrastructure

**Motivation**
[KTOR-7846](https://youtrack.jetbrains.com/issue/KTOR-7846) Fix running tests against Java 8 on CI

**Solution**
Fix problems making tests incompatible with Java 8 and remove `testJdk.coerceAtLeast(11)` from tests toolchain configuration.

_It is better to review commit-by-commit_

### CODE REVIEW

File: ktor-server/ktor-server-plugins/ktor-server-i18n/jvm/test/io/ktor/tests/i18n/TranslationTest.kt [hunk 1]
```diff
@@ -107,4 +109,10 @@ class TranslationTest {
         val contentAsString = response.bodyAsText()
         assertEquals("Русский Ключ", contentAsString)
     }
+
+    /**
+     * On Java 8 PropertyResourceBundle requires properties to be encoded in ISO-8859-1,
+     * while starting from Java 9 the default encoding is UTF-8.
+     */
+    private fun String.fixJava8Encoding(): String = this.toByteArray(Charsets.ISO_8859_1).toString(Charsets.UTF_8)
```
Comment (osipxd) [inline ktor-server/ktor-server-plugins/ktor-server-i18n/jvm/test/io/ktor/tests/i18n/TranslationTest.kt]: Should this "fix" be applied inside `i18n` implementation? Or we assume that users know that bundles should be in a different encoding on Java 8?

File: ktor-utils/jvm/src/io/ktor/util/NioPath.kt [hunk 2]
```diff
@@ -18,7 +19,11 @@ public fun Path.combineSafe(relativePath: Path): Path {
     }
     check(!normalized.isAbsolute) { "Bad relative path $relativePath" }
 
-    return resolve(normalized)
+    // On JVM 8 ZipPath implementation misses this check,
+    // so we have to check the path is not empty to not get an `ArrayIndexOutOfBoundsException`
+    if (this.nameCount == 0) return normalized
+
+    return this.resolve(normalized)
```
Comment (e5l) [inline ktor-utils/jvm/src/io/ktor/util/NioPath.kt:26]: it looks like we can drop `this`

File: ktor-server/ktor-server-plugins/ktor-server-i18n/jvm/test/io/ktor/tests/i18n/TranslationTest.kt [hunk 3]
```diff
@@ -107,4 +109,17 @@ class TranslationTest {
         val contentAsString = response.bodyAsText()
         assertEquals("Русский Ключ", contentAsString)
     }
+
+    /**
+     * On Java 8 PropertyResourceBundle requires properties to be encoded in ISO-8859-1,
+     * while starting from Java 9 the default encoding is UTF-8.
+     */
+    private fun String.fixJava8Encoding(): String {
+        return if (System.getProperty("java.version").startsWith("1.8")) {
```
Comment (e5l) [inline ktor-server/ktor-server-plugins/ktor-server-i18n/jvm/test/io/ktor/tests/i18n/TranslationTest.kt:118]: please consider extracting to the top level field

**Review by e5l — APPROVED** [review]
Looks good!

**Pull request merged**